training.py

```diff
@@ -0,0 +1,75 @@
+import argbind
+from datetime import datetime
+from pathlib import Path
+import time
+from torch.utils.data import DataLoader
+import torch.optim as optim
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+import data
+from model import Scribe
+import utils
+
+@argbind.bind(without_prefix=True)
+def train(
+        save_path: Path = Path('runs') / datetime.now().isoformat(),
+        num_training_iterations: int = 1000,
+        batch_size: int = 40,
+        random_seed: int = None):
+
+    """Trains model, samples an output, and displays it
+
+    Parameters
+    ----------
+    save_path : Path
+        The location of the tensorboard run (defaults to runs/CURDATETIME)
+    """    
+    writer = SummaryWriter(save_path)
+    if random_seed:
+        print(f"Setting random seed: {random_seed}")
+        torch.manual_seed(random_seed)
+    dataset = data.HandwritingDataset()
+    dl = DataLoader(dataset, shuffle=True, batch_size=batch_size, collate_fn=data.collate_fn)
+    inf = utils.infinite_dl(dl)
+
+    model = Scribe()
+    optimizer = optim.Adam(model.parameters())
+
+    iteration_times = []
+    for iteration, batch in enumerate(dl):
+        texts, texts_mask, strokes, strokes_mask = batch
+        if iteration >= num_training_iterations:
+            break
+        optimizer.zero_grad()
+        start = time.time()
+        output = model(strokes)
+        iteration_times.append(time.time() - start)
+
+        # output[0] is the prediction, strokes[1] is the ground truth
+        loss = ((strokes[1:] - output[:-1]) ** 2).mean()
```

Comment:
This looks good, but maybe in general I'd prefer to use the losses from the `torch.nn` library.